Test/filter snips

cathmer opened the pull request:
Write tests for filter snips class. Uses the annotation to access private methods, so these methods could be unit tested.

cathmer committed:
Comment filter snips

cathmer committed:
Set sequence length for nodes

Add tests for filter snips

cathmer committed:
Override hashcode of node

cathmer committed:
Delete Graph class

wouterjsmit commented:
The tests look good and comments are clear.

Could you squash the `Fix import order` 508f529 and `Put && on newline` 33c2fcc? These do not need to be separate commits. (squash them into the commit where you originally changed that line)

Could you also mention in the commit message of `Set sequence length for nodes` 3f6d191 and `Delete Graph class` bf8f76c why you did what you did? I'm not sure why the Graph class could be deleted or what the stray sequence length did.

I'm sure I'll hear if you you're having Git trouble :stuck_out_tongue:

FarisElghlan commented:
Next time try to build the project locally before commiting so you have fewer failing builds in a row. The tests look nice.

FarisElghlan commented on PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java:
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
+    assertTrue(root.getOutlinks().contains(9));
+
+    assertEquals(filteredGraph.getNode(8), graph.getNode(8));
+    assertEquals(filteredGraph.getNode(9), graph.getNode(9));
```
Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getNode` separately).

FarisElghlan commented on PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java:
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
     OriginalGraph filteredGraph = filter.filter();
-    // filteredGraph.print();
+    
+    assertEquals(filteredGraph.getSize(), 3);
```
The first parameter of `assertEquals` should be the expected value and the second parameter should be the actual value:
`assertEquals(3, filteredGraph.getSize());`

This comment also applies to a lot of your other assert statements

FarisElghlan commented on PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java:
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
+    assertEquals(snip.getInlinks().size(), 0);
+    
+    assertEquals(snip.getId(), 1);
+    assertEquals(snip.getSnips(), 2);
```
Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getInlinks` and `getId/getSnips` separately).

FarisElghlan commented on PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java:
```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
+    assertTrue(snip.getInlinks().contains(3));
+    
+    assertEquals(snip.getId(), 4);
+    assertEquals(snip.getSnips(), 1);
```
Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getInlinks` and `getId / getSnips` separately).

cathmer committed:
Fixed assertEquals.

Changed the order of the parameters in assertEquals, so the first input is the expected value and the second input is the real value.

cathmer commented:
@Pathemeous @FarisElghlan I squashed some commits together to make it look cleaner. I've also fixed the assertEquals, so that the first input is the expected value and the second input is the real value.

I didn't make more test methods to split up the asserts of a method, because I feel this is redundant and a waste of time, because the tests do check what they have to check, and it works. 

Also, the last commit has no value, but was used to check if it would make Travis build. It turned out Travis just needed some time (like an hour) to start building..

wouterjsmit commented:
@cathmer That sometimes happens, because Travis is free and their resources aren't endless. Very weird though.

Lastly, could I ask you to remove the last commit, since I really do not want that in the history.
Can be done with rebasing (remove the whole line for that commit)
or by resetting the tip of the branch one commit back:
`git reset --hard HEAD~1`

Followed by a `git push -f` to update the remote.

Git note:
One last thing that I'm also just noticing now. When you update dev into your local branch with a merge, you get an empty merge commit (because there were no conflicts and it could be fast-forwarded).
Locally, if you instead rebase your local branch onto `dev` (taking the tip of `dev` and addding your commits to there, rather than where you branched of before), this commit disappears (even after you have already done the merge.

I would ask you to rebase to remove the empty merge commit from `dev`, but I'm interested to see if github automatically removes that commit when this PR is merged.

cathmer commented:
The empty commit has been removed. I think it's ready for merging now.

wouterjsmit commented:
@cathmer Not sure what happened, but now I see some commits from `dev` on top of your changes.

Edit: If I check out your branch and perform the following two commands, I get the desired result, where only your changes are put on top of `dev`.
`git fetch origin && git rebase origin/dev`

cathmer commented:
Done, now the other commits are gone.
